### PR TITLE
Results page

### PR DESCRIPTION
**Pull request by yuval-ya**

### CODE REVIEW

File: evo-delivery-app/evo-delivery-frontend/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ In the project directory, you can run:
 
 ### `npm start`
 
-Runs the app in the development mode.\
+Results the app in the development mode.\
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/README.md:11]: ?

File: ea_server/README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ _Table of Contents:_
     - [Creation](#creation)
     - [Activation](#activation)
     - [Package management](#package-management)
-  - [Run](#run)
+  - [Result](#run)
```
Comment (tamirho) [inline ea_server/README.md:10]: ?

File: ea_server/README.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ Install new package (using the virtualenv pip):
 $ virtualenv/bin/pip install {package name}
 ```
 
-## Run
+## Result
```
Comment (tamirho) [inline ea_server/README.md:64]: ?

File: evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx [hunk 4]
```diff
@@ -17,8 +17,8 @@ export const LeafletMap = ({
   return (
     <MapContainer center={center} zoom={zoom} scrollWheelZoom={scrollWheelZoom} zoomControl={false} {...props}>
       <TileLayer
-        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-        url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
+        attribution='<a href="https://www.maptiler.com/copyright/" target="_blank">&copy; MapTiler</a> <a href="https://www.openstreetmap.org/copyright" target="_blank">&copy; OpenStreetMap contributors</a>'
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx:20]: Could you please explain what this change means?

File: evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx [hunk 5]
```diff
@@ -17,8 +17,8 @@ export const LeafletMap = ({
...
+        url='https://api.maptiler.com/maps/pastel/256/{z}/{x}/{y}.png?key=DZhCtHuWh6SXRgnbthLh'
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx:21]: Also

File: evo-delivery-app/evo-delivery-frontend/src/features/map/drawers/ResultsDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        routes?.flatMap((driverRoute) => (
+            driverRoute.orders.map((order) => (
+                <OrderMarker order={order} key={order._id}/>
+            ))
+        ))
+    );
+
+    const renderDepotMarker = () => (
+        <DepotMarker depot={depot as Depot} key={depot?._id}/>
+    );
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/features/map/drawers/ResultsDrawer.tsx:25]: Maybe we should consider extracting this logic to the runs page (or however that updates the state).

I mean that the map doesn't know the connections between route, order, and depot.
We will provide 3 different entities and the map will draw them for us.

File: evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Result/DriverRouteListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+                type: mapActions.UPDATE_CENTER,
+                payload: {center: [order.latitude, order.longitude] as LatLngTuple}
+            });
+            dispatch({
+                type: mapActions.UPDATE_ZOOM,
+                payload: {zoom: 13}
+            });
+        },
+        []
+    );
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Result/DriverRouteListItem.tsx:45]: try to use this hook - `useFocusLocation`
and update zoom and center together

File: evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Result/Result.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    useEffect(() => {
+        if (result) {
+            dispatch({
+                type: mapActions.UPDATE_RESULTS,
+                payload: {result: result, zoom: 13, }
+            });
+            dispatch({
+                type: mapActions.UPDATE_CENTER,
+                payload: {center: [result.depot?.latitude, result.depot?.longitude] as LatLngTuple }
+            });
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Result/Result.tsx:27]: try to update state with one action.. you can use 

`{ type: mapActions.UPDATE_STATE, payload: { center, result, zoom } }`

File: evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Results/Results.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+import RouteIcon from '@mui/icons-material/Route';
+import {EvaluateResult} from "../../../../../../evo-delivery-backend/src/types";
+import {useGetEntities} from "../../../../hooks/entities-api/use-get-entities";
+
+
+export const Results = () => {
+    const {data: results = [], isFetching, isLoading, isError} = useGetEntities();
+    const navigate = useNavigate();
+    const location = useLocation();
+    const goToEntity = useMemo(() => (id: string) => navigate(`${id}/${ENTITY_VIEW_STATES.view}${location.search}`), []);
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/results/components/Results/Results.tsx:16]: try to use this hook -> `useNavigateToChild`

File: evo-delivery-app/evo-delivery-frontend/src/pages/results/components/hooks/use-result.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import { useQuery } from '@tanstack/react-query';
+
+export const useResult = (resultId: string) => {
+    return useQuery(['results', resultId], () => fetch(`api/v1/results/${resultId}`).then(res => res.json()))
+}
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/results/components/hooks/use-result.ts:5]: you can replace that with -> `useGetEntity` hook

File: evo-delivery-app/evo-delivery-frontend/src/pages/results/components/hooks/use-results.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import { useQuery } from '@tanstack/react-query';
+
+export const useResults = () => {
+    return useQuery(['results'], () => fetch(`api/v1/results`).then(res => res.json()))
+}
```
Comment (tamirho) [inline evo-delivery-app/evo-delivery-frontend/src/pages/results/components/hooks/use-results.ts:5]: you can replace that with -> `useGetEntities` hook

**Review by tamirho — CHANGES_REQUESTED** [review]

File: evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx (hunk 4 above)
Comment (yuval-ya) [reply evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx:20]: see https://cloud.maptiler.com/maps/pastel/I forgot I changed it, we can keep the original if you didn't like

File: evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx (hunk 5 above)
Comment (yuval-ya) [reply evo-delivery-app/evo-delivery-frontend/src/features/map/LeafletMap.tsx:21]: See above

File: evo-delivery-app/evo-delivery-frontend/src/features/map/drawers/ResultsDrawer.tsx (hunk 6 above)
Comment (yuval-ya) [reply evo-delivery-app/evo-delivery-frontend/src/features/map/drawers/ResultsDrawer.tsx:25]: isn't it what it already does?

Comment (yuval-ya) [reply evo-delivery-app/evo-delivery-frontend/src/features/map/drawers/ResultsDrawer.tsx:25]: maybe i didn't understand

**Pull request merged**